libraries/AP_BattMonitor/AP_BattMonitor_ESC.cpp

```diff
@@ -92,4 +93,15 @@ void AP_BattMonitor_ESC::read(void)
     }
 }
 
+bool AP_BattMonitor_ESC::reset_remaining(float percentage)
+{
+    delta_mah = 0.0f;
+    read();
+    const float current_mah = _state.consumed_mah;
+    bool reset = AP_BattMonitor_Backend::reset_remaining(percentage);
+    delta_mah = _state.consumed_mah - current_mah;
```

Comment:
Presumably this should only be doineif the backend reset worked? (It does always work at the moment admittedly)

Reply:
ok fixed